Source/SwiftLintFramework/Rules/SortedImportsRule.swift

```diff
@@ -0,0 +1,57 @@
+//
+//  SortedImportsRule.swift
+//  SwiftLint
+//
+//  Created by Scott Berrevoets on 12/15/16.
+//  Copyright © 2016 Realm. All rights reserved.
+//
+
+import Foundation
+import SourceKittenFramework
+
+public struct SortedImportsRule: ConfigurationProviderRule, OptInRule {
+    public var configuration = SeverityConfiguration(.warning)
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "sorted_imports",
+        name: "Sorted Imports",
+        description: "Imports should be sorted.",
+        nonTriggeringExamples: [
+            "import AAA\nimport BBB\nimport CCC\nimport DDD"
+        ],
+        triggeringExamples: [
+            "import AAA\nimport ZZZ\nimport ↓BBB\nimport CCC"
+        ]
+    )
+
+    public func validateFile(_ file: File) -> [StyleViolation] {
+        let pattern = "import\\s+\\w+"
```

Comment:
To be completely honest, for me personally the hit on readability wouldn't be worth the benefits in this case:

- Not needing `previousMatch` is definitely nice, no argument there
- The bridging is reduced to once for every `import` - which is a decent win but probably not very noticeable considering the matches are short and there aren't many of them
- I wasn't aware of `matchPattern(_:withSyntaxKind:)`, but it appears that would even work with the current implementation (which I agree would definitely be nicer!)
- I personally don't have a huge problem with the `defer`, but I can see how that can require a double-take

That said, I'm totally onboard with following the general coding practices of the existing codebase (which your proposal seems more in line with) so I've pushed another commit with that implementation.